pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -0,0 +1,206 @@
+	opspec "github.com/rook/rook/pkg/operator/ceph/spec"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	mgrDaemonCommand = "/usr/bin/ceph-mgr"
+)
+
+func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) *extensions.Deployment {
+
+	podSpec := v1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   mgrConfig.ResourceName,
+			Labels: c.getDaemonLabels(mgrConfig.DaemonName),
+			Annotations: map[string]string{"prometheus.io/scrape": "true",
+				"prometheus.io/port": strconv.Itoa(metricsPort)},
+		},
+		Spec: v1.PodSpec{
+			InitContainers: []v1.Container{
+				// Config file init performed by Rook
+				c.makeConfigInitContainer(mgrConfig, opspec.ConfigInitContainerName),
+			},
+			Containers: []v1.Container{
+				// c.mgrContainer(mgrConfig.ResourceName, mgrConfig.DaemonName),
+				c.makeMgrDaemonContainer(mgrConfig, "mgr"),
+			},
+			RestartPolicy: v1.RestartPolicyAlways,
+			Volumes:       opspec.PodVolumes(""),
+			// Volumes: []v1.Volume{
+			// 	{Name: k8sutil.DataDirVolume, VolumeSource: v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}},
+			// 	k8sutil.ConfigOverrideVolume(),
+			// },
+			HostNetwork: c.HostNetwork,
+		},
+	}
+	if c.HostNetwork {
+		podSpec.Spec.DNSPolicy = v1.DNSClusterFirstWithHostNet
+	}
+	c.placement.ApplyToPodSpec(&podSpec.Spec)
+
+	replicas := int32(1)
+	d := &extensions.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mgrConfig.ResourceName,
+			Namespace: c.Namespace,
+		},
+		Spec: extensions.DeploymentSpec{Template: podSpec, Replicas: &replicas},
+	}
+	k8sutil.SetOwnerRef(c.context.Clientset, c.Namespace, &d.ObjectMeta, &c.ownerRef)
+	return d
+}
+
+func (c *Cluster) makeConfigInitContainer(mgrConfig *mgrConfig, containerName string) v1.Container {
+	return v1.Container{
+		Name: containerName,
+		Args: []string{
+			"ceph",
+			mgrdaemon.InitCommand,
+			fmt.Sprintf("--config-dir=%s", k8sutil.DataDir),
+			fmt.Sprintf("--mgr-name=%s", mgrConfig.DaemonName),
+		},
+		Image: k8sutil.MakeRookImage(c.Version),
+		Env: []v1.EnvVar{
+			{Name: "ROOK_MGR_KEYRING",
+				ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{Name: mgrConfig.ResourceName},
+						Key:                  keyringName,
+					}}},
+			k8sutil.PodIPEnvVar(k8sutil.PrivateIPEnvVar),
+			k8sutil.PodIPEnvVar(k8sutil.PublicIPEnvVar),
+			opmon.ClusterNameEnvVar(c.Namespace),
+			opmon.EndpointEnvVar(),
+			opmon.SecretEnvVar(),
+			opmon.AdminSecretEnvVar(),
+			k8sutil.ConfigOverrideEnvVar(),
+		},
+		VolumeMounts: opspec.RookVolumeMounts(),
+		// config file creation does not require ports to be open
+		Resources: c.resources,
+	}
+}
+
+func (c *Cluster) makeMgrDaemonContainer(mgrConfig *mgrConfig, containerName string) v1.Container {
+	return v1.Container{
+		Name: containerName,
```

Comment:
That's definitely an option. I opted to have it as a parameter so that the list of (Init)Containers in the pod spec would show the pod names all together in one place. The mon files give a better representation of my thoughts: https://github.com/rook/rook/blob/5d735cfc07e1f89b4a5424a90e3f41274b5e33b9/pkg/operator/ceph/cluster/mon/spec.go#L111

Thinking about this more, having non-parameterized funcs makes more sense. It would still be possible to locate the container names close together by making `const` definitions for the names at the top of the file. I'm not sure how helpful that would be though. Thoughts?

Reply:
Since they're just used once, I would lean towards including them inline. But if you find it useful, sounds good to declare them as consts next to each other at the top.